docs/examples.rst

```diff
@@ -4,4 +4,391 @@
+
+1. (Optional) ``keyboard-state`` branch of Nengo GUI. This will allow you to
+   control the GUI simulation using key presses.
+
+.. should we include instructions on installing a dev version of the GUI
+   and how to switch branches? or just leave it for people who know what
+   they are doing and hopefully it gets merged soon.
+
+To run this example in Nengo, from the |gui_dir| directory, run:
+
+.. code-block:: none
+
+   nengo 04-RL_demo.py -b nengo_fpga
+
+When the model is loaded into Nengo GUI, you should see a graphical
+representation similar to the image below.
+
+   .. image:: _static/rl_gui.png
+      :alt: RL Example GUI
+      :width: 100%
+
+For this example, the graphical pane is divided into four quadrants.
+
+- In the top left quadrant, the network diagram for this example is displayed.
+- In the bottom left quadrant, the values for the radar (``stim_radar``) and
+  the reward values associated with the agent's current action (``reward``) are
+  shown.
+- A graphical representation of the world map, and the agent's current location
+  in this map is displayed in the top right quadrant. The agent is represented
+  as a blue triangle, and the obstacles in the world with black squares. When
+  the agent encounters an obstacle, it changes from a blue to a red colour.
+  Also indicated is the current "learning" state of the agent -- the agent
+  is learning when **Explore** is **ON**, and conversely, is not adapting to
+  the reward signals when **Explore** is **OFF**.
+- The bottom right quadrant displays the output of the action selection network
+  (``Thalamus``), as well as the associated spike rasters for each action
+  (move forward: ``actions_0``; turn counter-clockwise: ``actions_1``;
+  turn clockwise: ``actions_2``)
+
+When the simulation is running, the agent should move around the world, as
+shown in the images below. The images below also show the agent moving around
+the world in 3 modes:
+
+   1. Exploring: The agent is constantly using the error signal to learn
+      which is the best action to perform.
+   #. Exploiting: The learning has been disabled after the agent has spent
+      about 15 seconds in the "Exploring" mode. The agent is thus moving
+      about the world using the radar-to-action mapping it had learned in the
+      exploration phase.
+   #. No Learning: The simulation is started with learning disabled. In this
+      scenario, the agent moves forward, encounters a wall, and is unable to
+      recover.
+
+   .. list-table::
+      :widths: 33 33 33
+      :align: center
+
+      * -
+
+         .. do not delete this table row
+
+        -
+        -
+
+      * -
+
+         .. image:: _static/rl_explore.gif
+
+         |nbsp| |nbsp| |nbsp| Exploring
+
+        -
+
+         .. image:: _static/rl_exploit.gif
+
+         |nbsp| |nbsp| |nbsp| Exploiting
+
+        -
+
+         .. image:: _static/rl_no-learn.gif
+
+         |nbsp| |nbsp| |nbsp| No Learning
+
+Changing the World Layout

```

Comment:
In `nengo-pynq-hw` we don't do full capitalization on 4th level headings. Do we want to fully capitalize this or only the first word?